pycalphad/io/tdb.py

```diff
@@ -417,13 +407,16 @@ def _setitem_raise_duplicates(dictionary, key, value):
 
 def to_interval(relational):
     if isinstance(relational, And):
-        return Intersection(*[to_interval(i) for i in relational.args])
+        result = UniversalSet()
+        for i in relational.args:
+            result = result.intersection(to_interval(i))
+        return result
     elif isinstance(relational, Or):
         return Union(*[to_interval(i) for i in relational.args])
     elif isinstance(relational, Not):
         return Complement(*[to_interval(i) for i in relational.args])
     if relational == S.true:
-        return Interval(S.NegativeInfinity, S.Infinity, left_open=True, right_open=True)
+        return Interval(S.NegativeInfinity, S.Infinity, True, True)
```

Comment:
Can you open an issue/PR in symengine.py to keep compatibility with sympy?

Reply:
https://github.com/symengine/symengine.py/issues/380